Tests/Packet++Test/Tests/NflogTests.cpp

```diff
@@ -0,0 +1,37 @@
+#include "../TestDefinition.h"
+#include "../Utils/TestUtils.h"
+#include "EndianPortable.h"
+#include "MacAddress.h"
+#include "Packet.h"
+#include "NflogLayer.h"
+#include "IPv4Layer.h"
+#include "SystemUtils.h"
+
+
+PTF_TEST_CASE(NflogPacketParsingTest)
+{
+	timeval time;
+	gettimeofday(&time, NULL);
+
+	READ_FILE_AND_CREATE_PACKET_LINKTYPE(1, "PacketExamples/nflogPacket.dat", pcpp::LINKTYPE_NFLOG);
+
+	pcpp::Packet nflogPacket(&rawPacket1);
+
+	PTF_ASSERT_TRUE(nflogPacket.isPacketOfType(pcpp::NFLOG));
+	PTF_ASSERT_EQUAL(nflogPacket.getFirstLayer()->getProtocol(), pcpp::NFLOG, enum);
+	pcpp::NflogLayer* nflogLayer = nflogPacket.getLayerOfType<pcpp::NflogLayer>();
+	PTF_ASSERT_NOT_NULL(nflogLayer->getNextLayer());
+
+	PTF_ASSERT_EQUAL(nflogLayer->getFamily(), pcpp::IPv4);
+	PTF_ASSERT_EQUAL(nflogLayer->getVersion(), 0);
+	PTF_ASSERT_EQUAL(be16toh(htobe16(nflogLayer->getResourceId())), 42);
```

Comment:
Since `getResourceId()` already use `htobe16()` the check here should be simpler:
```cpp
PTF_ASSERT_EQUAL(nflogLayer->getResourceId(), 42);
```